src/Compilers/CSharp/Test/Semantic/Semantics/ColorColorTests.cs

```diff
@@ -1729,6 +1733,49 @@ class X
             CompileAndVerify(comp, expectedOutput: "42");
         }
 
+        [WorkItem(5362, "https://github.com/dotnet/roslyn/issues/5362")]
+        [Fact]
+        public void TestColorColorSymbolInfoInArrowExpressionClauseSyntax()
+        {
+            const string source = @"public enum Lifetime
+{
+    Persistent,
+    Transient,
+    Scoped
+}
+public class Example
+{
+    public Lifetime Lifetime => Lifetime.Persistent;
+    //                          ^^^^^^^^
+}";
+           CreateCompilationWithMscorlibAndSystemCore(source, options: TestOptions.ReleaseDll)
+                .VerifyAnalyzerOccurrenceCount(new[] { new ColorColorSymbolInfoInArrowExpressionClauseSyntaxAnalyzer() }, 0);
```

Comment:
So, if the HandleMemberAccessExpression is never called, for whatever reason, the test is going to pass? Perhaps the instance of ColorColorSymbolInfoInArrowExpressionClauseSyntaxAnalyzer can record the fact of HandleMemberAccessExpression being called, which the test can verify here?